client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/GetWatchResponse.java

```diff
@@ -0,0 +1,140 @@
+        this.status = status;
+        this.source = source;
+        this.xContentType = xContentType;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public long getVersion() {
+        return version;
+    }
+
+    public boolean isFound() {
+        return version != Versions.NOT_FOUND;
+    }
+
+    public WatchStatus getStatus() {
+        return status;
+    }
+
+    public BytesReference getSource() {
+        return source;
+    }
+
+    public XContentType getContentType() {
+        return xContentType;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        GetWatchResponse that = (GetWatchResponse) o;
+        return version == that.version &&
+            Objects.equals(id, that.id) &&
+            Objects.equals(status, that.status) &&
+            Objects.equals(source, that.source) &&
+            xContentType == that.xContentType;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, status, source, xContentType, version);
+    }
+
+    private static final ParseField ID_FIELD = new ParseField("_id");
+    private static final ParseField FOUND_FIELD = new ParseField("found");
+    private static final ParseField VERSION_FIELD = new ParseField("_version");
+    private static final ParseField STATUS_FIELD = new ParseField("status");
+    private static final ParseField WATCH_FIELD = new ParseField("watch");
+
+    private static ConstructingObjectParser<GetWatchResponse, Void> PARSER =
+        new ConstructingObjectParser<>("get_watch_response", true,
+            a -> {
+                boolean isFound = (boolean) a[1];
+                if (isFound) {
+                    XContentSource source = (XContentSource) a[4];
+                    return new GetWatchResponse((String) a[0], (long) a[2], (WatchStatus) a[3], source.source, source.contentType);
+                } else {
+                    return new GetWatchResponse((String) a[0]);
+                }
+            });
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), ID_FIELD);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), FOUND_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), VERSION_FIELD);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(),
+            (parser, context) -> WatchStatus.parse(parser), STATUS_FIELD);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(),
+            (parser, context) -> {
+                try (XContentBuilder builder = XContentBuilder.builder(parser.contentType().xContent())) {
+                    builder.copyCurrentStructure(parser);
+                    BytesReference ref = BytesReference.bytes(builder);
+                    return new XContentSource(ref, parser.contentType());
```

Comment:
Could we return the builder here instead of creating a new type `XContentSource`?